packages/flutter_tools/lib/src/macos/xcode.dart

```diff
@@ -287,7 +312,95 @@ class XCDevice {
     return null;
   }
 
-  List<dynamic> _cachedListResults;
+  /// Observe identifiers (UDIDs) of devices as they attach and detach.
+  ///
+  /// Each attach and detach event is a tuple of one event type
+  /// and identifier.
+  Stream<Map<XCDeviceEvent, String>> observedDeviceEvents() {
+    if (!isInstalled) {
+      _logger.printTrace("Xcode not found. Run 'flutter doctor' for more information.");
+      return null;
+    }
+    return _deviceIdentifierByEvent.stream;
+  }
+
+  // Attach: d83d5bc53967baa0ee18626ba87b6254b2ab5418
+  // Detach: d83d5bc53967baa0ee18626ba87b6254b2ab5418
+  final RegExp _observationIdentifierPattern = RegExp(r'^(\w*): (\w*)$');
+
+  Future<void> _startObservingTetheredIOSDevices() async {
+    try {
+      if (_deviceObservationProcess != null) {
```

Comment:
should this be `== null` ? This gets called from the `listen` from `_setupDeviceIdentifierByEventStream`, but the whenDone nulls out the process: https://github.com/flutter/flutter/pull/58137/files#diff-ebc648b2067d73f2af30539f5d3a59faR389

Reply:
It should always be null since as you saw the `whenDone` should handle nulling it out.
https://github.com/flutter/flutter/pull/58137#discussion_r432661646
> Can you check/assert somehow that you're not blowing away your handle to an active process?

I think the concern is there would be a logic error where `_startObservingTetheredIOSDevices` could be called twice without the process exiting.

Reply:
Oops, I forgot how != worked briefly...